js/gestionPresupuesto.js

```diff
@@ -1,25 +1,124 @@
-// TODO: Crear las funciones, objetos y variables indicadas en el enunciado
+var presupuesto = 0;
+var gastos = new Array();
+var idGasto = 0;
 
-// TODO: Variable global
+function actualizarPresupuesto(nuevoPresupuesto) {
+    if(nuevoPresupuesto >= 0 && typeof nuevoPresupuesto == 'number'){
+        presupuesto = nuevoPresupuesto;
+        return presupuesto;
+    }
+    else{
+        return -1;
+    }
+}
+
+function mostrarPresupuesto() {
+    return `Tu presupuesto actual es de ${presupuesto} €`;
+}
 
+function CrearGasto(descripcion, valor, fecha, ...etiquetas) {
+    this.etiquetas = [];
```

Comment:
En lugar de esto, puedes utilizar `this.anyadirEtiquetas`. De esta manera te evitas crear un gasto con etiquetas duplicadas.